### PR TITLE
[tests-only][full-ci] merge `test-acceptance-api` and `test-acceptance-from-core-api` commands

### LINKED ISSUE #9874: [QA] single command to run all API/CLI tests
https://github.com/owncloud/ocis/issues/9872#issuecomment-2301257071
>One suggestion: maybe we can join `make test-acceptance-from-core-api` to `make test-acceptance-api`?
>but keep the old set naming (prefixes with `core`) to use only one command to run tests: 

Currently, we have two different commands to run the ocis API tests:

- `make test-acceptance-api`: to run API/CLI tests for local ocis tests
- `make test-acceptance-from-core-api`: ro run API tests taken from oc10 repo

It would be better to have a single command to run all the tests. So ti achieve this we can do the following things:
- [x] remove command [test-acceptance-from-core-api](https://github.com/owncloud/ocis/blob/ff0ac7e06999196f6668778d40db3826d38bbf0e/Makefile#L136-L138)
- [x] merge behat [configs](https://github.com/owncloud/ocis/tree/master/tests/acceptance/config) `behat.yml` and `behat-core.yml`
- [x] use command `test-acceptance-api` for all tests
- [ ] update the [docs](https://owncloud.dev/ocis/development/testing/)

### PR DESCRIPTION
**Pull request by ghost**
## Description

This PR
- removes command `test-acceptance-from-core-api`
- merge behat configs `behat.yml` and `behat-core.yml`
- uses command `test-acceptance-api` for running all tests

## Related Issue

- Fixes: https://github.com/owncloud/ocis/issues/9874

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -176,9 +176,40 @@ config = {
...
+        "numberOfParts": 28,
         "skip": False,
         "skipExceptParts": [],
+        "suites": [
+            "coreApiMain",
+            "coreApiAuth",
+            "coreApiCapabilities",
+            "coreApiFavorites",
+            "coreApiShareCreateSpecialToShares1",
+            "coreApiShareCreateSpecialToShares2",
```
Comment (saw-jan) [inline .drone.star:188]: is it not possible to run suites in parts?

File: Makefile [hunk 2]
```diff
@@ -125,17 +125,11 @@ clean-tests:
...
 PARALLEL_BEHAT_YML=tests/parallelDeployAcceptance/config/behat.yml
-# behat config file for core api tests
-CORE_BEHAT_YML=tests/acceptance/config/behat-core.yml
 
 .PHONY: test-acceptance-api
 test-acceptance-api: vendor-bin/behat/vendor
 	BEHAT_BIN=$(BEHAT_BIN) $(PWD)/tests/acceptance/run.sh
 
-.PHONY: test-acceptance-from-core-api
-test-acceptance-from-core-api: vendor-bin/behat/vendor
```
Comment (saw-jan) [inline Makefile:136]: remove the command from above help info as well

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Review by saw-jan — CHANGES_REQUESTED** [review]
I see remnants of `test-acceptance-from-core-api` in `tests/acceptance/docker`. Please, fix them as well.
Also, the docker compose setup should work.

And do the ocis bump in reva

File: tests/acceptance/config/behat.yml [hunk 3]
```diff
@@ -424,6 +424,332 @@ default:
...
+          ldapUsersOU: TestUsers
+          ldapGroupsOU: TestGroups
+          ldapInitialUserFilePath: /../../config/ldap-users.ldif
+      contexts:
+        - FeatureContext: &common_feature_context_params
+            baseUrl: http://localhost:8080
+            adminUsername: admin
+            adminPassword: admin
+            regularUserPassword: 123456
+            ocPath: apps/testing/api/v1/occ
```
Comment (phil-davis) [inline tests/acceptance/config/behat.yml:442]:
```suggestion
        - FeatureContext: *common_feature_context_params
```

The FeatureContext params are already like this at line 19. I think that we don't need to repeat them here.

File: tests/acceptance/docker/src/run-tests.sh [hunk 4]
```diff
@@ -9,6 +9,7 @@ git config --global advice.detachedHead false
 
 if [ "$TEST_SOURCE" = "oc10" ]
 then
+    export ACCEPTANCE_TEST_TYPE='core-api'
```
Comment (phil-davis) [inline tests/acceptance/docker/src/run-tests.sh:12]: what uses this code path?
do we still have test runs that have `"$TEST_SOURCE" = "oc10"` ?

File: tests/acceptance/config/behat.yml [hunk 5]
```diff
@@ -424,6 +424,332 @@ default:
...
 
+    coreApiMain:
+      paths:
+        - "%paths.base%/../features/coreApiMain"
+      context: &common_ldap_suite_context
+        parameters:
+          ldapAdminPassword: admin
+          ldapUsersOU: TestUsers
+          ldapGroupsOU: TestGroups
+          ldapInitialUserFilePath: /../../config/ldap-users.ldif
```
Comment (saw-jan) [inline tests/acceptance/config/behat.yml:435]:
```suggestion
      context: *common_ldap_suite_context
```

File: tests/acceptance/docker/src/run-tests.sh (hunk 4 above)
Comment (phil-davis) [reply tests/acceptance/docker/src/run-tests.sh:12]: Probably this whole `"$TEST_SOURCE" = "oc10" ` "if" block is not used any more?
I suppose it was used for parallel deployment testing, and I think that we have removed the code for that.
But we can clean up old oarallel deployment and code that tries to mention "oc10" in another cleanup PR. That is not really related to the refactoring work of this PR.

**Review by phil-davis — APPROVED** [review]
Seems OK now, if the `reva` run also works fairly easily with this test code refactoring.

**Review by saw-jan — COMMENTED** [review]
Please, update the docs as well

File: tests/acceptance/docker/src/run-tests.sh (hunk 4 above)
Comment (prashant-gurung899) [reply tests/acceptance/docker/src/run-tests.sh:12]: > Probably this whole `"$TEST_SOURCE" = "oc10" ` "if" block is not used any more? I suppose it was used for parallel deployment testing, and I think that we have removed the code for that. But we can clean up old oarallel deployment and code that tries to mention "oc10" in another cleanup PR. That is not really related to the refactoring work of this PR.

I can to this clean up of unnecessary code in my PR: https://github.com/owncloud/ocis/pull/9900

**Review by saw-jan — APPROVED** [review]
LGTM :+1: 
@nirajacharya2 Please, follow up the docs changes

**Pull request merged**